azure/resource-containers/pf-resource-group/classes/pf-resource-group-options.ps1

```diff
@@ -1,8 +1,8 @@
-Class PfResourceGroupOptions {
+Class PfResourceGroupOptions : PfResourceContainer {
     [string]$Name
     [string]$SubscriptionName
     [AzRegion]$AzRegion
-    [hashtable]$Tags
+    [System.Collections.Generic.Dictionary[string,string]]$Tags = [System.Collections.Generic.Dictionary[string,string]]::new()
     [AzLock]$Lock
```

Comment:
So az-modules take hashtable objects for tags.  However, I found convert-tojson won't serialize this object like you said because it is not strongly typed: https://stackoverflow.com/questions/27939880/serialize-hashtable-using-json-net

I have now converted these to string dictionaries.  I haven't tested the convertto-json functionality yet, but I have tested the conversion back to hashtable and it works fine, or should for az-modules

Reply:
I'm assuming somewhere now there is some kind of import of the system.collections.generic.dictionary based on .net core installed on machine?  Going to assume anyone running PS 7 is going to have *some* kind of .net core though, so i imagine not going to be possible cause of issue

Reply:
i should add cool solution =)

hopefully json plays along - i imagine this not too diff than using a ps object with fields under hood though so solid chance...

Reply:
Yeah, there was no import for me on my personal laptop after installing powershell 7.  So whether or not powershell brought those generic libraries in or it's something that was already on my machine is tough to know.  When I get the foundation solid, I would like to test the json before moving onto connecting resources to azure.  Right now I am trying to get the base code pretty solid so I can work on dependencies and not have to refactor a bunch of stuff after the hard part is done.  I added json export to our project board.

>